File: code/leetcode/4.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        from heapq import merge
        sortedli = list(merge(nums1, nums2))
        mid = len(sortedli)//2
        if mid % 2 == 0:
            x, y = sortedli[mid - 1], sortedli[mid]
            z = float(x + y)
            return z/2
        else:
            return sortedli[mid]

File: code/cpp/leetcode/4.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> result;
        result.reserve(nums1.size()+nums2.size());
        merge(nums1.begin(),nums1.end(),nums2.begin(),nums2.end(),back_inserter(result));
        return(result.size()%2==0)?(result[(result.size()/2)-1]+result[result.size()/2])/2.0:result[result.size()/2];
    }
};